numpy/ma/core.py

```diff
@@ -6077,9 +6074,6 @@ def max(self, axis=None, out=None, fill_value=None, keepdims=np._NoValue):
             if result.ndim:
                 # Set the mask
                 result.__setmask__(newmask)
-                # Get rid of Infs
-                if newmask.ndim:
-                    np.copyto(result, result.fill_value, where=newmask)
```

Comment:
We have a few PRs all in the same space here, I think?  It would be helpful to see how they overlap.

I am not very comfortable with just removing this.  While I don't like it, I think downstream may have some expectations on the masked values.

Reply:
> I am not very comfortable with just removing this. While I don't like it, I think downstream may have some expectations on the masked values.

While I understand your concerns about removing this, I believe it might be more beneficial to adhere to the documentation and use the output of `minimum_fill_value`. This approach could provide clarity and consistency, which may ultimately align better with user expectations.

Reply:
Actually, there's another PR aimed at fixing a different bug that involves storing the `fill_value` as `np.uint64`, which could also resolve the issue. Additionally, simply adding `_check_fill_value` before `copyto` could serve as a solution as well. However, I believe the current approach is preferable because it aligns better with the documentation.

Reply:
That is fair and I am not always un-inclined to making improvements rather than just fixing a bug.

But in this case, I would feel better to concentrate on fixing the bug, rather than fixing the bug and also adding an API tweak/enhancement (even if that tweak may be small).

If the another PR is sufficient to fix the bug here, then this will simply become an API or ENH PR.

Reply:
I understand your perspective. I have reverted the changes and focus solely on fixing the bug.